src/librustc_driver/driver.rs

```diff
@@ -577,15 +577,13 @@ pub fn phase_2_configure_and_expand<'a>(sess: &Session,
     //
     // baz! should not use this definition unless foo is enabled.
 
-    let mut feature_gated_cfgs = vec![];
-    krate = time(time_passes, "configuration 1", || {
-        sess.track_errors(|| {
-            syntax::config::strip_unconfigured_items(sess.diagnostic(),
-                                                     krate,
-                                                     sess.opts.test,
-                                                     &mut feature_gated_cfgs)
-        })
-    })?;
+    krate = time(time_passes, "configuration", || {
+        let (krate, features) =
+            syntax::config::strip_unconfigured_items(krate, &sess.parse_sess, sess.opts.test);
+        // these need to be set "early" so that expansion sees `quote` if enabled.
+        *sess.features.borrow_mut() = features;
+        krate
+    });
```

Comment:
I don't see any reason to stop compilation -- the only errors that get reported here are malformed `cfg` attributes (in which case the item is kept), malformed `cfg_attr` attributes (in which case the attribute is removed), and malformed `feature` attributes (in which case we might not see the features).
At worst, the user might see some gated feature errors for a feature they intended to use.